sdk/formrecognizer/ai-form-recognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,131 @@
+
+### Build model error
+
+When building a model, the most common errors are caused by an [invalid data set](#invalid-training-data-set) or an [invalid SAS Url](#invalid-sas-url).
+
+#### Invalid training data set
+
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set [on Microsoft Learn by following this link](https://aka.ms/customModelV3).
+
+Example error below:
+
+```
+ RestError: Invalid request.
+  {
+   "name": "RestError",
+   "code": "InvalidRequest",
+   "statusCode": 400,
+   .
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidRequest",
+       "message": "Invalid request.",
+       "innererror": {
+         "code": "InvalidContent",
+         "message": "The file is corrupted or format is unsupported. Refer to documentation for the list of supported formats."
+       }
+     }
+   },
+   "message": "Invalid request."
+ }
+```
+
+#### Invalid SAS URL
+
+This error points to missing permissions on the blob storage SAS URL for the Form Recognizer service to access the training dataset resource. For more information about SAS tokens for Form Recognizer, see [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/create-sas-tokens).
+
+Example error:
+
+```
+ RestError: Invalid argument.
+  {
+   "name": "RestError",
+   "code": "InvalidArgument",
+   "statusCode": 400,
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidSasToken",
+         "message": "The shared access signature (SAS) is invalid: SAS token expired on 10/04/2022 03:19:05 +00:00"
+       }
+     }
+   },
+   "message": "Invalid argument."
+ }
+```
+
+### Generic Error
+
+Seeing a "Generic error" returned from the SDK is most often caused by heavy load on the service. For troubleshooting issues related to service limits, see related information [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/service-limits?tabs=v30).
+
+Example error:
+
+```
+(3014) Generic error during training.
+Invalid model created with ID=<model ID>
+```
+
+### Unexpected time to build a custom model
+
+Training(`DocumentModelAdministrationClient#beginBuildDocumentModel()`) with the "neural" build mode typically takes significantly longer than the "template" mode.
+
+For simpler use-cases, you can use the `template` build mode which uses a different model building algorithm that takes less time. Neural models are trained using deep learning, so they are more computationally intensive to train and use. More information about the `template` and neural build modes and the characteristics of models created using each can be found at the following Microsoft Learn pages:
+
+- [DocumentModelBuildMode](https://docs.microsoft.com/javascript/api/@azure/ai-form-recognizer/documentmodelbuildmode)
```

Comment:
nit: this isn't the same kind of thing as the other two bullets. I'd put it after the two bullet points as a note and say something like 

```
_See also_: [`DocumentModelBuildMode` reference documentation](https://docs.microsoft.com/javascript/api/@azure/ai-form-recognizer/documentmodelbuildmode).
```